### ISSUE TITLE
client Sync should exclude learner endpoint

### ISSUE THREAD

**Issue by yishuT** [etcd-io/etcd#13853]
### What happened?

We see `etcdserver: rpc not supported for learner` error from clients with AutoSync enabled.  https://github.com/etcd-io/etcd/blob/main/client/v3/client.go#L188 here does not exclude learner endpoint

### What did you expect to happen?

Either requests are not sent to learner endpoints only or learner endpoints proxy requests to non learner member

### How can we reproduce it (as minimally and precisely as possible)?

Add a new member but do not promote it, and wait for client autosync to pick the new member

### Etcd version (please run commands below)

etcdctl version: 3.5.0
API version: 3.5

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands blow, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by serathius** [comment]
There is a PR in flight https://github.com/etcd-io/etcd/pull/13837

**Comment by serathius** [comment]
Fix will be available in v3.5.3
